### DISCUSSION TITLE
Native way for pagination API response

### DISCUSSION THREAD

**Discussion by makarovstas** [prkumar/uplink#286] [Q&A]
Hi [prkumar](https://github.com/prkumar), please give advice for this case: some APIs have a paginated response, so, we have to make several requests with same type in recursion mode, in which only one parameter changes (offset, sinceID and etc). What do you think, is there a way to implement this inside Consumer class, or is it better to make it a separate functionality?

**Comment by wabiloo** [comment]
Same question - and to make it more complex, the pagination information is returned as Response Headers...
